src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/scheduling/GlobalConfChecker.java

```diff
@@ -132,6 +137,38 @@ private void checkGlobalConf() {
         } catch (Exception e) {
             throw translateException(e);
         }
+
+        if (SystemProperties.getAutoUpdateTimestampServiceUrl()) {
+            updateTimestampServiceUrls(serverConf);
+        }
+    }
+
+    private void updateTimestampServiceUrls(ServerConfType serverConf) {
```

Comment:
Since this method, and different conflict / duplicate name cases that could happen are pretty complex, I would **strongly** recommend adding an automatic test for this method (probably calls for some refactoring to be able to test)

Reply:
Added a test case in the commit https://github.com/nordic-institute/X-Road/pull/877/commits/d19990101271d8521fa33ba9beae701d43389bc7